### PR TITLE
Reexport ink_metadata in transcoder crate

### LINKED ISSUE #1521: InkProject not exported from ink_metadata
https://github.com/paritytech/cargo-contract/blob/bcc1c8cc5a748b3f421f40b9d9a9d7d355c8acb2/crates/transcode/src/lib.rs#L177

In the transcode crate, the InkProject type is currently unexported, making it inaccessible to users. It would be beneficial to expose the ink_metadata module entirely, allowing users to include it in their projects without the need to explicitly add the ink_metadata crate solely for accessing certain types.

We propose updating the lib.rs file as follows:
```rust
// Other imports
pub use ink_metadata;
...
```

This change would enable users to access the necessary types conveniently.

### PR DESCRIPTION
**Pull request by 0xLucca**
## Summary
Closes #1521 
- [n] y/n | Does it introduce breaking changes?
- [n] y/n | Is it dependent on the specific version of `ink` or `pallet-contracts`?

## Description

This PR solves issue #1521

## Checklist before requesting a review
- [x] My code follows the style guidelines of this project
- [x] I have added an entry to `CHANGELOG.md`
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Export ink_metadata types
```
Comment (cmichi) [inline CHANGELOG.md:9]:
```suggestion

### Added
- Export `ink_metadata` types in `transcode` crate - [#1522](https://github.com/paritytech/cargo-contract/pull/1522)
```

**Comment by cmichi** [comment]
@0xLucca Can you fix the CI? It should be a simple `cargo fmt`.

image

File: crates/transcode/src/lib.rs [hunk 2]
```diff
@@ -124,10 +124,26 @@ use anyhow::{
     Context,
     Result,
 };
-use ink_metadata::{
-    ConstructorSpec,
+pub use ink_metadata::{
```
Comment (ascjones) [inline crates/transcode/src/lib.rs:127]: Why not just reexport the whole crate i.e. `pub use ink_metadata`?

Gives us namespacing and means we don't have to add/remove types.

Comment (0xLucca) [reply crates/transcode/src/lib.rs:127]: Reexporting the whole crate works for me

Comment (cmichi) [reply crates/transcode/src/lib.rs:127]: If you update the PR today we can add it to an upcoming release.

Comment (0xLucca) [reply crates/transcode/src/lib.rs:127]: I can do that

**Review by ascjones — APPROVED** [review]

**Comment by ascjones** [comment]
Please provide a more descriptive title to the PR

**Review by cmichi — APPROVED** [review]

**Pull request merged**